src/fsipy/simulations/offset_stenosis.py

```diff
@@ -218,22 +188,21 @@ def create_bcs(t, v_, DVP, mesh, boundaries, domains, mu_f,
     # Assemble boundary conditions
     bcs = u_inlet + [d_inlet, u_inlet_s, d_inlet_s, d_rigid]
 
-    # Define the pressure condition (apply to inner surface, numerical instability results from applying to outlet)
-    #dso = ds(outlet_id1, domain=mesh, subdomain_data=boundaries) # Outlet surface # Maybe try applying to all outlets???
-
-
-
-    # Load fourier coefficients and scale by flow rate
+    # Load fourier coefficients for the pressure and scale by flow rate
     An_P, Bn_P = np.loadtxt(os.path.join(os.path.dirname(os.path.abspath(__file__)), P_FC_File)).T
 
+    # Apply pulsatile pressure at the fsi interface by modifying the variational form
+    n = FacetNormal(mesh)
     dSS = Measure("dS", domain=mesh, subdomain_data=boundaries)
     p_out_bc_val = InnerP(t=0.0, t_ramp=0.2, An=An_P, Bn=Bn_P, period=T_Cycle, P_mean=P_mean, degree=p_deg)
-    F_solid_linear += p_out_bc_val * inner(n('+'), psi('+'))*dSS(fsi_id)  # defined on the reference domain
+    F_solid_linear += p_out_bc_val * inner(n('+'), psi('+')) * dSS(fsi_id)
```

Comment:
I'm not sure this is safe. As I mentioned in https://github.com/KVSlab/turtleFSI/issues/40 one-sided interior facet integrals needs a volume subdomain to be correctly oriented.

Reply:
I see. We do have subdomain tags. Fluid domain is always marked as 1 and the solid domain is marked as 2. When constructing `F`, we do not have whole domain integral like `*dx` but we do have integral over each domain https://github.com/KVSlab/turtleFSI/blob/34925bed03008e463ce6ddc82b875287fc866177/turtleFSI/modules/fluid.py#L32 and https://github.com/KVSlab/turtleFSI/blob/34925bed03008e463ce6ddc82b875287fc866177/turtleFSI/modules/solid.py#L40. Is that enough to ensure the consistent orientation?

Reply:
As long as `dx` is a ufl Measure with a subdomain_marker with this orientation in it, any integral should be fine.

Reply:
Great! Then, I think we are safe here. We have `dx_f[fluid_region] = dx(dx_f_id[fluid_region], subdomain_data=domains)` where `domains` contain subdomain tags for the fluid and solid (1 and 2).

Reply:
Add that and multiply by a 0 constant, and then things should be consistently ordered only using parameters from the cells with marker 2. (ie normals point out of material 2)

Reply:
We do have other terms with `dx` inside `F_solid_linear` as you can see https://github.com/KVSlab/turtleFSI/blob/34925bed03008e463ce6ddc82b875287fc866177/turtleFSI/modules/solid.py#L39-L41 (for example, `(rho_s/k * inner(v_["n"] - v_["n-1"], psi)*dx_s[solid_region]`) 

Shouldn’t that be enough to make sure the consistent orientation? When I add any term with `Constant(0)*dx`, then there is arity mismatch error. 

```
raise ArityMismatch("Adding expressions with non-matching form arguments {0} vs {1}.".format(_afmt(a), _afmt(b)))
ufl.algorithms.check_arities.ArityMismatch: Adding expressions with non-matching form arguments ('v_0',) vs ().
```